### PR TITLE
TS-4623 Adds a pre-commit script for git

### PR DESCRIPTION
**Pull request by zwoop**

### CODE REVIEW

**Comment by zwoop** [comment]
Need input from the git expert, too bad @bgaff doesn't have a computer still :-/. So will have to settle for @jpeach and @PSUdaemon .

**Comment by atsci** [comment]
FreeBSD build *successful*! See https://ci.trafficserver.apache.org/job/Github-FreeBSD/392/ for details.

**Comment by atsci** [comment]
Linux build *successful*! See https://ci.trafficserver.apache.org/job/Github-Linux/288/ for details.

File: tools/pre-commit [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+ROOT=$(git rev-parse --show-toplevel)/.git/fmt
+case $(uname -s) in
+Darwin)
+    FORMAT=${FORMAT:-${ROOT}/clang-format/clang-format.osx}
+    ;;
+Linux)
+    FORMAT=${FORMAT:-${ROOT}/clang-format/clang-format.linux}
+    ;;
+*)
+    echo "Leif needs to build a clang-format for $(uname -s)"
```
Comment (jpeach) [inline tools/pre-commit]: Just run $(git rev-parse --show-toplevel)/tools/clang-format.sh here ?

**Comment by jpeach** [comment]
👍

File: tools/pre-commit (hunk 1 above)
Comment (zwoop) [inline tools/pre-commit]: Hmmm, I copied / borrowed that from your clang-format.sh script ...

**Pull request closed without merging**